extension/src/popup/components/account/AccountAssets/index.tsx

```diff
@@ -20,44 +24,71 @@ const AssetIcon = ({
 }) =>
   assetIcons[code] || code === "XLM" ? (
     <img
-      className="AssetWrapper--asset-logo"
+      className="AccountAssets--asset-logo"
       alt={`${code} logo`}
       src={code === "XLM" ? StellarLogo : assetIcons[code] || ""}
       onError={() => {
         retryAssetIconFetch({ key: issuerKey, code });
       }}
     />
   ) : (
-    <div className="AssetWrapper__asset-bullet" />
+    <div className="AccountAssets__asset-bullet" />
   );
 
 export const AccountAssets = ({
-  assetIcons,
+  assetIcons: inputAssetIcons,
   sortedBalances,
-  retryAssetIconFetch,
 }: {
   assetIcons: AssetIcons;
   sortedBalances: Array<any>;
-  retryAssetIconFetch: (arg: { key: string; code: string }) => void;
-}) => (
-  <div className="AssetWrapper">
-    {sortedBalances.map(({ token: { issuer, code }, total }) => (
-      <div className="AssetWrapper__asset" key={code}>
-        <div className="AssetWrapper__copy-left">
-          <AssetIcon
-            assetIcons={assetIcons}
-            code={code}
-            issuerKey={issuer?.key}
-            retryAssetIconFetch={retryAssetIconFetch}
-          />
-          <span>{code}</span>
-        </div>
-        <div className="AssetWrapper__copy-right">
-          <div>
-            {new BigNumber(total).toString()} <span>{code}</span>
+}) => {
+  const [assetIcons, setAssetIcons] = useState(inputAssetIcons);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const [hasIconFetchRetried, setHasIconFetchRetried] = useState(false);
+
+  const retryAssetIconFetch = async ({
```

Comment:
moved the `retryAssetIconFetch` method to inside `AccountAssets`, so that don't have to re-define when using

Reply:
Good call - it looks like we don't even need `assetIcons` at all in `views/Account/index.tsx`. Should we move all that logic here?

Reply:
ya  I can do that, do you mind if I do in a next PR? it'll take some refactoring in views/Account

Reply:
Yeah, totally.

Actually, if it'll be a bit of a lift, let's ignore for now and focus on Send Payment. We can come back to this later

Reply:
sounds good, ya I think in this refactor we'd probably want to go further down the rabbit hole and also refactor `getAssetIcons` to not depend on Balances, I'll create a ticket to do later 👍 
https://github.com/stellar/freighter/issues/404

Reply:
94951a70c4198e4b9ef10ce0fc5cd195c6c5a2be